revscoring/dependencies/tests/test_context.py

```diff
@@ -0,0 +1,34 @@
+from nose.tools import eq_
+
+from ..context import Context
+from ..dependent import Dependent
+
+
+def test_context():
+    # No context
+    context = Context()
+    foo = Dependent("foo", lambda: "foo")
+    bar = Dependent("bar", lambda: "bar")
+    foobar = Dependent("foobar", lambda foo, bar: foo + bar,
+                       depends_on=[foo, bar])
+    eq_(context.solve(foobar), "foobar")
+    eq_(list(context.solve([foo, bar, foobar])), ["foo", "bar", "foobar"])
+
+    # Cache context
+    context = Context(cache={bar: "baz"})
+    eq_(context.solve(foobar), "foobaz")
+
+    # Context context
```

Comment:
It does.  I nearly called this "ContextManager" but it seemed too verbose.  

I sort of think the conflation of terms is OK because they mean the same thing and serve the same purpose.  Once is a "call context" and the other is a "preserved context".  Thoughts?